red_october/README.rst

```diff
@@ -0,0 +1,201 @@
+  * When in demo mode, a sample encrypted field is available under the `Preferences`
+    tab in the `res.users` form view
+
+If there is a value saved in the encrypted field, the user will be immediately prompted
+for their decryption password.
+
+If there is no value in the field, the user is not prompted for a password until they
+edit the record, add a value, then save the record.
+
+Once a password is entered, it will be used to decrypt all encrypted fields until the
+user refreshes their browser.
+
+The user that initially adds the encrypted value will be the owner of the file. They will
+need to subsequently grant ownership others if they also need access to the data.
+
+Note that there must be a valid rights delegation on the Vault in order to decrypt a file,
+even if you are the only owner.
+
+Delegate Decryption Rights
+--------------------------
+
+Proper rights delegation is required to allow the Red October Vault to decrypt data using
+your credentials.
+
+Once decryption rights have expired, a Vault will no longer be able to decrypt your data,
+even with the correct password. Adding decryption rights to the Vault again will restore
+its ability to decrypt your data.
+
+Note that a Vault can only maintain one delegation at a time per user, so any new delegations
+will replace existing delegations for the same user.
+
+* Click the `Crypto Menu`
+* Click the name of the Vault you want to delegate rights to
+* Enter the delegation information:
+  * Number of Uses - How many decryptions are allowed for this delegation
+  * Date of Expiration - When will this delegation expire, if the Number of Uses has not been
+    exceeded
+  * Password - Password to the currently active Crypto Profile
+
+Change Password
+---------------
+
+New Crypto Profiles are created using the current user's Odoo password, which is stored as
+plain text in the Odoo request session. This is insecure, and it is recommended that all
+users immediately change their password before working with sensitive data.
+
+* Click the `Crypto Menu`
+* Click `Change Password`
+* Enter the requested information
+
+Try It
+------
+
+.. image:: https://odoo-community.org/website/image/ir.attachment/5784_f2813bd/datas
+   :alt: Try me on Runbot
+   :target: https://runbot.odoo-community.org/runbot/149/10.0 for server-tools
+
+Known issues / Roadmap
+======================
+
+* Add caching for most methods
+* Allow transferring files between vaults.
+* Add delegation uses & delta to ``red.october.file.owner``.
+* Proper handling for multiple profiles per user
+* All model methods that touch the password or decrypted data need to be moved from
+  Odoo models into standard objects. This will provide more security, because the Odoo
+  model inheritance magic can be utilized to inject malicious code with a module. Using
+  the standard Python inheritance will require that malicious code is placed directly in
+  the base module in order to be used.
+* Audit Javascript XSS vulnerabilities
```

Comment:
I'll keep this in mind - at the moment I'm not too worried, but an external audit will need to happen as we start scaling. Thanks!